File: ios/Runner/BridgeHandlers/NativeChannelHandler.swift
import Flutter
import UIKit

class NativeChannelHandler {

    func handle(call: FlutterMethodCall, result: @escaping FlutterResult) {

        switch call.method {

        case "openPhoneDialer":
            PhoneDialerHandler.open(call: call, result: result)

        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

//
//  NativeChannelHandler.swift
//  Runner
//
//  Created by Alkaff M1 on 10/04/26.
//

File: android/app/src/main/kotlin/com/example/native_phone_dialer/NativeChannelHandler.kt

package com.example.native_phone_dialer.channel

import android.app.Activity
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class NativeChannelHandler(
    private val activity: Activity
) : MethodChannel.MethodCallHandler {

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {

        when (call.method) {

            "openPhoneDialer" -> {
                val phone = call.argument<String>("phoneNumber") ?: ""
                PhoneDialerHandler.open(activity, phone)
                result.success("Dialer opened")
            }

            else -> result.notImplemented()
        }
    }
}